sdk/keyvault/keyvault-certificates/src/identifier.ts

```diff
@@ -4,26 +4,16 @@
 import { parseKeyvaultIdentifier } from "./generated/utils";
 
 /**
- * Valid collection names for Key Vault Certificate identifiers.
+ * Represents a Key Vault Certificate Id and its parsed contents.
  */
-export type KeyVaultCertificatesIdentifierCollectionName = "certificates" | "deletedcertificates";
-
-/**
- * Represents a Key Vault identifier and its parsed contents.
- */
-export interface ParsedKeyVaultCertificatesIdentifier {
-  /**
-   * The type of resource under Key Vault that this identifier is referring to.
-   */
-  collection: KeyVaultCertificatesIdentifierCollectionName;
-
+export interface KeyVaultCertificateId {
   /**
-   * The originally received identifier.
+   * The source ID that was parsed.
```

Comment:
Since one can create a json object following this interface on their own without using our parse function, we can consider not tying ourselves to the parse function either. In .Net, this is not an issue as the parse function is on the struct that represents the parsed output, so the two are already coupled.

Use of an example can make it clear that this is a url. Other languages wont have this problem as they are not using string

```suggestion
   * The complete representation of the Key Vault Certificate Id like (add an example here)
```

Reply:
I pushed a fix for this! Let me know what you think.